components/sensor/as5600.rst

```diff
@@ -0,0 +1,307 @@
+  in the neighborhood of ``2100`` for a well positioned magnet.
+
+  - **name** (**Required**, string): The name for the magnitude sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+- **status** (*Optional*): Information about the magnet status. Typically for diagnostic purposes.
+  See :ref:`Magnet Status <as5600-magnet-status>`
+
+  - **name** (**Required**, string): The name for the status sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+- All other options from :ref:`Sensor <config-sensor>`.
+
+
+.. _as5600-out-of-range-mode:
+
+Out of Range Mode
+*****************
+
+- ``min_max`` (default)
+- ``nan``
+
+When using a range less that 360deg, there would be a range of raw values that would be considered "out of range". By default (``min_max`` mode), the AS5600
+splits that range in half and reports ``0`` while in the half of the "out-of-range" range closest to the ``start_position`` and it reports
+``4095`` while in the half of the "out-of-range" range closest to the ``end_position`` / end of the ``range``. Alternatively, you may set to ``nan``
+mode where the sensor will publish ``NAN`` (i.e. "Unknown") when the position falls outside the narrowed range.
+
+.. _as5600-magnet-status:
+
+Magnet Status
+*************
+
+The magnet status should report one of the following values:
+
+- ``2`` indicates that no magnet was detected.
+- ``4`` indicates that the magnet was detected and has good reading.
+- ``5`` indicates that the magnet was detected, but is too strong. Measurements may appear to be stuck if the magnet is too strong.
+- ``6`` indicates that the magnet was detected, but is too weak. Measurements may still be possible in this state.
+
+.. _as5600-converting-position:
+
+Converting Position
+*******************
+
+You may desire the position to be converted from the native ``0`` thru ``4095`` to degrees, or perhaps a percentage of the allowable range.
+Here are some examples to make that happen:
+
+.. figure:: images/as5600-ui-derived.jpg
+    :align: center
+    :width: 80.0%
+
+.. code-block:: yaml
+
+    as5600:
+      id: my_as5600
+    sensor:
+      - platform: as5600
+        update_interval: 1s
+        name: Angle
+        unit_of_measurement: '°'
+        accuracy_decimals: 2
+        icon: mdi:rotate-right
+        filters:
+          - delta: 1
+          - lambda: 'return x * as5600::RAW_TO_DEGREES * id(my_as5600).get_range_scale();'
+      - platform: as5600
+        update_interval: 1s
+        name: Percent
+        unit_of_measurement: '%'
+        accuracy_decimals: 2
+        icon: mdi:rotate-right
+        filters:
+          - delta: 1
+          - lambda: 'return (x / as5600::POSITION_COUNT) * 100;'
```

Comment:
Instead of suggesting more sensors which will increase the communications to the chip, it would be better to define a regular position sensor, then use `copy` sensors with the filters to adjust. This is more efficient as it updates instantly when the source sensor is updated.

Reply:
Correct

Reply:
So while I think the copy sensor is valuable to know. I'm glad to know it exists now, but I think what I am going to do here is actually just split the 2 sensors to 2 separate configuration samples. The intent is really just to show how to convert the position value to an angle or percentage. I would imagine in many cases people would be using one or the other. And if I add extra fluff that is really only needed because I tried to combine 2 different examples would ultimately be bad because I assume a good majority of the time people that may not have a deeper understanding end up copying the code blocks into their config and hope it works. Then I imagine people would just remove the sensor they don't want, and then they have extra sensors that they don't technically need.